File: Factory Method/FactoryMethod - Java/principal/src/main/java/com/formatfactory/pagamento/model/CartaoDeCredito.java
package com.formatfactory.pagamento.model;

import org.springframework.stereotype.Component;

@Component("CartaoDeCredito")
public class CartaoDeCredito implements Pagamento{
    @Override
    public void realizaPagamento(Double valor) {
        System.out.println("Pagando " + valor + " usando cartão de crédito.");
    }    
}

File: Factory Method/FactoryMethod - PHP/src/model/CartaoDeCredito.php
<?php

require_once "Pagamento.php";

class CartaoDeCredito implements Pagamento{
    public function realizaPagamento(Float $valor){
        echo "Pagamento de " . number_format($valor, 2, ',', '.') . " realizado utilizando Cartão de Crédito.";
    }
}

?>